### DISCUSSION TITLE
Setting up Spotify

### DISCUSSION THREAD

**Discussion by sumnerboy12** [micro-nova/AmpliPi#210] [Q&A] [answered]
Hi,

I just received my AmpliPi all the way down here in NZ!! I am having a play around and trying to setup my Spotify account (premium family) as I would like to set up a separate account per input, so the kids can play their playlists in their rooms etc.

But editing the pre-configured Spotify stream, or trying to add a new one, doesn't have any configuration options (in the web UI) and I can't find any mention in the docs about how to do this.

Is this feature implemented yet? Is it a file based config I need to update somewhere?

Cheers,
Ben

**Comment by linknum23** [comment, marked as answer]
Sorry if it's confusing, but at the moment out Spotify streams don't take user account info. This allows multiple accounts to be able to play off of the same source.

For your use case I would suggest making a uniquely named Spotify stream per user. Then select each as an input to a different source #.

**Reply by marc-portier** [reply]
@linknum23 -- Sorry for re-adressing this issue, but I'm obviously missing something here. 
How do I go about "making a Spotify stream per user" 

To be clear: the only spotify-setting-related screen I seem to find is this:
![image](https://user-images.githubusercontent.com/299574/150864731-539b0736-1274-4a67-9e37-fdf39a051b5f.png)

There is a single "Spotify Device" already - but nowhere to add account credentials? 
What am I missing? 
(am now even confused enough to doubt if I should be looking for other amplipi screens or over in spotify land?) 

> coin dropped some time later -- these devices created on the amplipi end up being picked up as devices your spotify app can detect as targets to send the music to.

**Reply by Lohrer** [reply]
You can create as many Spotify 'Streams' as you want as long as you give them a different name. So as an example, from the page in your screenshot you could name the first 'Spotify Device' Stream type as 'AmpliPi' and click Add Stream. Then click Add New Stream again and name a second 'Spotify Device' 'AmpliPi2'. Now click the 1 tab and change it to 'AmpliPi - spotify'. Click the 2 tab and change it to 'AmpliPi2 - spotify'. Finally if you open spotify on your phone or desktop and click the Connect to Device button you should see both AmpliPi and AmpliPi2 as devices to play to.

There are no credentials required on AmpliPi because it's just showing up on your network as a device capable of connecting to. The logging in to Spotify part happens on your phone or desktop or whatever device you use to initiate the Spotify song playback.

**Reply by marc-portier** [reply]
But also your intended default behavior is not ringing any bells here -- can't seem to parse "This allows multiple accounts to be able to play off of the same source." 

Should my spotify app discover the amplipi device? 
> found this out 15' later:
> ANSWER: yes it does, but only if it is actually active on a tab

**Reply by linknum23** [reply]
Exactly! Like all other streams, a Spotify stream is only active/discoverable when it is connected to one of the input sources. I understand that this can be a little confusing. There is probably a way that we can document this functionality better, but we haven't figured out where it would make sense yet.

**Reply by marc-portier** [reply]
Well, maybe inline, i.e. in the UI is the most appropriate place?

--> suggestion: a help icon linking to a managebale online resource that can be managed independently and eventually even include embedded youtube tutorials  (just make proper URi designed endpoints like https://help.amplipi.com/{version}/topic/streams/spotify into some decent online documentation management system (could be github/gitpages based and co-managed by the community)

--> on the other hand: now that I see how elegant it all works, I must have been totally thickheaded (and stubborn at it) not to realize the proper operation - given that observation I would probably not have noticed any help icon either :thinking:

**Comment by sumnerboy12** [comment]
Thanks, that makes sense, I should have tried that!

So I set up 4 Spotify streams on my AP, one for each user, but when I open
my Spotify app on my phone and try to connect to them to send a playlist, I
can only see the first stream, that was pre configured out of the box.

That first stream works fine but have your tested with multiple streams?

Sorry for the noob questions.

**Reply by linknum23** [reply]
The Spotify streams are only available when they are selected as an input. Go to each on of the 1-4 input tabs and select a different Spotify stream in each.

**Reply by sumnerboy12** [reply]
Ah yes - sorry - that is working now. Apologies for the silly questions!

**Reply by linknum23** [reply]
No silly questions. These will help us focus our documentation (like the quick start guide [here](https://github.com/micro-nova/AmpliPi/blob/main/docs/QUICK_START.md)) and inform future interface designs.

If you have any ideas on what we should change in the UI feel free to leave feedback here: #34

**Reply by linknum23** [reply]
Also we have noticed that a Spotify stream can get in a bad state and stop responding/playing, this is likely due to a bug in the underlying Spotify client, Vollibrespot.  If you see this you can always select a different input like the "             " input and then select the Spotify stream again. This sequence will restart the stream.

**Comment by sumnerboy12** [comment]
24hrs later and now I am seeing two of my AmpliPi streams in the Spotify app. Is there a limit on what the app can handle does anyone know?

**Reply by linknum23** [reply]
We have tested it with 4 Spotify streams. They seemed to work fine.
